File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/application/command/WithdrawUserService.java
package dev.starryeye.book_library.application.command;

import dev.starryeye.book_library.domain.user.User;
import dev.starryeye.book_library.domain.user.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Transactional
@Service
@RequiredArgsConstructor
public class WithdrawUserService {

    private final UserRepository userRepository;

    public void withdraw(Long id) {
        User user = userRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("user is not found, id = " + id));

        userRepository.delete(user);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/application/user/command/WithdrawUserService.kt
package dev.starryeye.book_library.application.user.command

import dev.starryeye.book_library.domain.user.UserRepository
import dev.starryeye.book_library.util.findByIdOrThrow
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Transactional
@Service
class WithdrawUserService(
    private val userRepository: UserRepository,
) {

    fun withdraw(id: Long) {

        val user = userRepository.findByIdOrThrow(id, "user is not found, id = $id")

        userRepository.delete(user)
    }
}